From c4a598686a1c6517c4c1a6c35e94fde1b38d9e49 Mon Sep 17 00:00:00 2001
From: Michael Adam <obnox@samba.org>
Date: Fri, 11 Dec 2015 00:37:13 +0100
Subject: [PATCH 195/214] hook-scripts: fix S30Samba scripts on systemd systems

/etc/init.d/smb does not exist on systemd systems.
Using "service smb <COMMAND>" is the portable way.
It calls init scripts on sysv systems, and redirects
to systemctl on systemd systems.

Change-Id: I7146c9998a51d8b170d3321f3477e92704ae7615
BUG: 1292762
Signed-off-by: Michael Adam <obnox@samba.org>
Reviewed-on: http://review.gluster.org/12945
Reviewed-on: http://review.gluster.org/13000
Tested-by: Gluster Build System <jenkins@build.gluster.com>
Reviewed-by: Anoop C S <anoopcs@redhat.com>
Tested-by: NetBSD Build System <jenkins@build.gluster.org>
Reviewed-by: Niels de Vos <ndevos@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/64209
Reviewed-by: Rajesh Joseph <rjoseph@redhat.com>
Tested-by: Rajesh Joseph <rjoseph@redhat.com>
---
 extras/hook-scripts/set/post/S30samba-set.sh     |    2 +-
 extras/hook-scripts/start/post/S30samba-start.sh |    2 +-
 extras/hook-scripts/stop/pre/S30samba-stop.sh    |    2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/extras/hook-scripts/set/post/S30samba-set.sh b/extras/hook-scripts/set/post/S30samba-set.sh
index e359059..97d067f 100755
--- a/extras/hook-scripts/set/post/S30samba-set.sh
+++ b/extras/hook-scripts/set/post/S30samba-set.sh
@@ -99,7 +99,7 @@ function sighup_samba () {
         then
                 kill -HUP "$pid";
         else
-                /etc/init.d/smb condrestart
+                service smb condrestart
         fi
 }
 
diff --git a/extras/hook-scripts/start/post/S30samba-start.sh b/extras/hook-scripts/start/post/S30samba-start.sh
index 43dc8e1..752eca6 100755
--- a/extras/hook-scripts/start/post/S30samba-start.sh
+++ b/extras/hook-scripts/start/post/S30samba-start.sh
@@ -82,7 +82,7 @@ function sighup_samba () {
         then
                 kill -HUP "$pid";
         else
-                /etc/init.d/smb condrestart
+                service smb condrestart
         fi
 }
 
diff --git a/extras/hook-scripts/stop/pre/S30samba-stop.sh b/extras/hook-scripts/stop/pre/S30samba-stop.sh
index 8950eea..62cf7d1 100755
--- a/extras/hook-scripts/stop/pre/S30samba-stop.sh
+++ b/extras/hook-scripts/stop/pre/S30samba-stop.sh
@@ -61,7 +61,7 @@ function sighup_samba () {
         then
                 kill -HUP $pid;
         else
-                /etc/init.d/smb condrestart
+                service smb condrestart
         fi
 }
 
-- 
1.7.1

